src/main/java/org/gradle/wrapper/GroovyWrapperMain.java

```diff
@@ -55,13 +54,23 @@ public static void main(String[] args) throws Exception {
 
         addSystemProperties(gradleUserHome, rootDir);
 
-        Logger logger = logger(options);
+        WrapperConfiguration config = WrapperExecutor.forWrapperPropertiesFile(propertiesFile).getConfiguration();
+        PathAssembler.LocalDistribution localDist = new PathAssembler(gradleUserHome).getDistribution(config);
+        File gradleHome = getAndVerifyDistributionRoot(localDist.getDistributionDir());
 
-        WrapperExecutor wrapperExecutor = WrapperExecutor.forWrapperPropertiesFile(propertiesFile);
-        wrapperExecutor.execute(
-                args,
-                new Install(logger, new Download(logger, "gradlew", wrapperVersion()), new PathAssembler(gradleUserHome)),
```

Comment:
I axed all this `Install` and `Download` invocation, but if you want to _keep_ the download-and-install functionality, you leave this in and just replace that `wrapperVersion()` with org.gradle.wrapper.Download.UNKNOWN_VERSION: https://github.com/gradle/gradle/blob/00862b8160a09051e8226736314afd16e7f418f7/subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java#L24

aka `"0"`